cql3/query_processor.cc

```diff
@@ -896,12 +903,23 @@ query_processor::execute_internal(
     }
 }
 
+// Explicit specializations for the execute_internal template
+template
+future<::shared_ptr<untyped_result_set>> query_processor::do_execute_internal(
+        const sstring& query_string,
+        db::consistency_level cl,
+        service::query_state& query_state,
+        const std::initializer_list<data_value>& values,
+        cache_internal cache);
+
+template <std::ranges::range Range>
+requires std::convertible_to<std::ranges::range_value_t<Range>, data_value>
 future<::shared_ptr<untyped_result_set>>
 query_processor::execute_with_params(
         statements::prepared_statement::checked_weak_ptr p,
         db::consistency_level cl,
         service::query_state& query_state,
-        const std::initializer_list<data_value>& values) {
+        const Range& values) {
```

Comment:
I think this could be done more cleanly by having a `data_value_list` type that's a wrapper over `std::vector<raw_value>`. Give that two non-explicit constructors - from `std::initializer_list<data_value>` and your range. Then execute_with_params() passes `data_value_list` to make_internal_options. This limits the templating to a small area.

Reply:
Sounds promising.  I'll try that